devices/lidl.js

```diff
@@ -140,6 +141,96 @@ const fzLocal = {
             }
         },
     },
+    watering_time_left: {
```

Comment:
Could you refactor this to a valueconverter? (such that we only use `tuya.tz.datapoints` for this device)

Reply:
Technically this should be possible, because this relies only on the ON/OFF DP and the current/previous state.
But then I need to replace the continues reporting of the remaining time by a JS interval. Is it possible to report state changes programmatically?

The current approach uses the continuous reporting of the battery level (every minute) to update the remaining time. But if there is way to do it programmatically, I can refactor this.


What I mean is something like that

```js
var timer;
function  valudeContverter() {
	if (ON) {
	  timer = setInterval(() => {
		// report state programmatically --> how?
		device.updateState({time_left: xxx});
		}, 1000 * 60);
	} else {
		clearInterval(timer);
		timer = undefined;
	}
}
```

Reply:
I found a `publish` method, which is used [in another converter](https://github.com/Koenkk/zigbee-herdsman-converters/blob/c45a74c630baa319c64b5574497db70eb35f6cd1/converters/fromZigbee.js#L502) to publish the state programmatically. This should work here too, if we pass it down to the `DP.from` converter.

@Koenkk does it makes sense this way?

Reply:
@zinserjan yes, feel free to pass the `publish` method

Reply:
Works now :blush: